### PR TITLE
dev: initialize submodules in `dev doctor`

### LINKED ISSUE #72247: dev: `dev doctor` should initialize submodules for you if necessary
Implement [this logic](https://github.com/cockroachdb/cockroach/blob/08734f16b79f8557109e90e376cee9f14fd7d062/Makefile#L72) in `dev doctor`.

Epic CRDB-8036

### PR DESCRIPTION
**Pull request by aayushshah15**
This commit adds a check to `dev doctor` to initialize submodules, like we do
in our `Makefile`.

Fixes #72247

Release note: None

### CODE REVIEW

File: pkg/cmd/dev/doctor.go [hunk 1]
```diff
@@ -99,6 +100,24 @@ Please perform the following steps:
...
+	const submodulesMarkerPath = binDir + "/.submodules-initialized"
+	d.log.Println("doctor: running submodules check")
+	if _, err := os.Stat(submodulesMarkerPath); errors.Is(err, os.ErrNotExist) {
+		if _, err = d.exec.CommandContextSilent(ctx, "git", "rev-parse", "--is-inside-work-tree"); err != nil {
+			return err
+		}
+		if _, err = d.exec.CommandContextSilent(ctx, "git", "submodule", "update", "--init", "--recursive"); err != nil {
+			return err
+		}
+		if _, err = d.exec.CommandContextSilent(ctx, "mkdir", "-p", binDir); err != nil {
```
Comment (rickystewart) [inline pkg/cmd/dev/doctor.go:113]: Nit: You can use `d.os.MkdirAll(binDir)` instead of shelling out to `mkdir`.

File: pkg/cmd/dev/doctor.go [hunk 2]
```diff
@@ -99,6 +100,24 @@ Please perform the following steps:
...
+			return err
+		}
+		if _, err = d.exec.CommandContextSilent(ctx, "touch", submodulesMarkerPath); err != nil {
```
Comment (rickystewart) [inline pkg/cmd/dev/doctor.go:116]: Nit: You can use `d.os.writeFile(submodulesMarkerPath, "")` instead of shelling out.

**Review by rickystewart — APPROVED** [review]

File: pkg/cmd/dev/doctor.go (hunk 1 above)
Comment (aayushshah15) [reply pkg/cmd/dev/doctor.go]: Done.

File: pkg/cmd/dev/doctor.go (hunk 2 above)
Comment (aayushshah15) [reply pkg/cmd/dev/doctor.go]: Done

**Comment by aayushshah15** [comment]
TFTR 

bors r+

**Pull request merged**